src/pretix/static/pretixcontrol/js/ui/checkinrules.js

```diff
@@ -0,0 +1,476 @@
+                        );
+                        return $ret;
+                    },
+                };
+            }
+        },
+        watch: {
+            placeholder: function (val) {
+                $(this.$el).empty().select2(this.opts());
+                this.build();
+            },
+            required: function (val) {
+                $(this.$el).empty().select2(this.opts());
+                this.build();
+            },
+            url: function (val) {
+                $(this.$el).empty().select2(this.opts());
+                this.build();
+            },
+        },
+        destroyed: function () {
+            $(this.$el)
+                .off()
+                .select2("destroy");
+        }
+    });
+
+    Vue.component('checkin-rule', {
+        template: ('<div v-bind:class="classObject">'
+            + '<div class="btn-group pull-right">'
+            + '<button type="button" class="checkin-rule-remove btn btn-xs btn-default" @click.prevent="wrapWithOR">OR</button>'
+            + '<button type="button" class="checkin-rule-remove btn btn-xs btn-default" @click.prevent="wrapWithAND">AND</button> '
+            + '<button type="button" class="checkin-rule-remove btn btn-xs btn-default" @click.prevent="cutOut" v-if="operands && operands.length == 1 && (operator === \'or\' || operator == \'and\')"><span class="fa fa-cut"></span></button>'
+            + '<button type="button" class="checkin-rule-remove btn btn-xs btn-default" @click.prevent="remove" v-if="level > 0"><span class="fa fa-trash"></span></button>'
+            + '</div>'
+            + '<select v-bind:value="variable" v-on:input="setVariable" required class="form-control">'
+            + '<option value="and">' + gettext('All of the conditions below (AND)') + '</option>'
+            + '<option value="or">' + gettext('At least one of the conditions below (OR)') + '</option>'
+            + '<option v-for="(v, name) in vars" :value="name">{{ v.label }}</option>'
+            + '</select> '
+            + '<select v-bind:value="operator" v-on:input="setOperator" required class="form-control" v-if="operator !== \'or\' && operator !== \'and\'">'
+            + '<option></option>'
+            + '<option v-for="(v, name) in operators" :value="name">{{ v.label }}</option>'
+            + '</select> '
+            + '<select v-bind:value="timeType" v-on:input="setTimeType" required class="form-control" v-if="vartype == \'datetime\'">'
+            + '<option value="date_from">' + gettext('Event start') + '</option>'
+            + '<option value="date_to">' + gettext('Event end') + '</option>'
+            + '<option value="date_admission">' + gettext('Event admission') + '</option>'
+            + '<option value="custom">' + gettext('custom time') + '</option>'
+            + '</select> '
+            + '<datetimefield v-if="vartype == \'datetime\' && timeType == \'custom\'" :value="timeValue" v-on:input="setTimeValue"></datetimefield>'
```

Comment:
I think this is a little bit misleading.

It should be either the datetime-picker but "custom date and time" or only a time-picker. The latter might make sense for a "3 day afternoon ticket - entry only starting at 3pm".